src/View/Helper/ListsHelper.php

```diff
@@ -297,24 +297,56 @@ public function displayDownloadLink($listId)
      *
      * @return void
      */
-    public function displayTranslationsDropdown($listId, $translationsLang = null) {
-        echo __('Show translations :') . ' ';
-
-        $path = $this->Url->build(['action' => 'show', $listId]) . '/';
-
-        // TODO onSelectedLanguageChange should be defined in a separate js file
-        echo $this->_View->element(
-            'language_dropdown',
-            array(
-                'name' => 'translationLangChoice',
-                'languages' => $this->Languages->languagesArrayShowTranslationsIn(),
-                'initialSelection' => $translationsLang,
-                'onSelectedLanguageChange' => "window.location.pathname = '$path' + language.code",
-                'forceItemSelection' => true,
-            )
-        );
-
+    public function displayTranslationsDropdown($listId,$filterLanguage, $translationsLang = null) {
+        ?>
+        <div class="section md-whiteframe-1dp">
+            <h2><?php echo __('Show translations in:'); ?></h2>
+            <?php
+            $path = $this->Url->build(['action' => 'show', $listId]) . '/';
+            // TODO onSelectedLanguageChange should be defined in a separate js file
+            echo $this->_View->element(
+                'language_dropdown',
+                array(
+                    'name' => 'translationLangChoice',
+                    'languages' => $this->Languages->languagesArrayShowTranslationsIn(),
+                    'initialSelection' => $translationsLang,
+                    'onSelectedLanguageChange' => "window.location.pathname = '$path' + '$filterLanguage' + '/'+language.code",
+                    'forceItemSelection' => true,
+                )
+            );
+            ?>
+            </div>
+        <?php
+    }
 
+    /**
+     * Display dropdown for selecting the language/s used to filter list
+     *
+     * @param [int] $listId 
+     * @param [string] $translationsLang Translation languages for each sentence
+     * @param [string] $filterLanguage Language whose sentences will only be displayed
+     * @return void
+     */
+    public function displayFilterByLangDropdown($listId, $filterLanguage, $translationsLang)
+    {
+        ?>
+        <div class="section md-whiteframe-1dp">
+            <h2><?php echo __('Sentences in:'); ?></h2>
+            <?php
+            $path = $this->Url->build(['action' => 'show', $listId]) . '/';
+            echo $this->_View->element(
+                'language_dropdown',
+                array(
+                    'name' => 'filterLanguageSelect',
+                    'languages' => $this->Languages->languagesArrayShowTranslationsIn(true, false),
+                    'initialSelection' => $filterLanguage,
```

Comment:
I think there is a contradiction here. You are filling a sentence language filter with choices from a list of languages for translations (languagesArrayShowTranslationsIn()). I think you could use languagesArrayAlone() instead and not add this new parameter for disabling the option "none".

Reply:
Thanks, had missed that function. Had initially found `onlyLanguagesArray()` . I think eventually some of these functions will need to be composed into fewer, more generalized functions. 

I imagine having a single `buildLanguageArray(params)` function that outputs a language list array used in views would be ideal.